knowledge-base/create-custom-image-properties-resolver-net-standard.md

```diff
@@ -0,0 +1,187 @@
+        }
+        catch (UnknownImageFormatException ex)
+        {
+            // Image format not recognized.
+            throw new NotSupportedException("Unsupported image format.", ex);
+        }
+
+        return size;
+    }
+
+    public override bool TryGetRawImageData(byte[] imageData, out byte[] rawRgbData, out byte[] rawAlpha, out TelerikSize size)
+    {
+        try
+        {
+            IImageFormat imageFormat;
+            using (ImageSharp image = ImageSharp.Load(imageData, out imageFormat))
+            {
+                size = new TelerikSize(image.Width, image.Height);
+
+                IImageDecoder decoder = null;
+                Dictionary<Type, Action> decoderSwitch = new Dictionary<Type, Action>
+                    {
+                        { typeof(PngFormat), () => decoder = new PngDecoder() },
+                        { typeof(BmpFormat), () => decoder = new BmpDecoder() },
+                        { typeof(GifFormat), () => decoder = new GifDecoder() },
+                        { typeof(JpegFormat), () => decoder = new JpegDecoder() },
+                        { typeof(PbmFormat), () => decoder = new PbmDecoder() },
+                        { typeof(TgaFormat), () => decoder = new TgaDecoder() },
+                        { typeof(TiffFormat), () => decoder = new TiffDecoder() },
+                        { typeof(WebpFormat), () => decoder = new WebpDecoder() },
+                    };
+
+                if (decoderSwitch.ContainsKey(imageFormat.GetType()))
+                {
+                    decoderSwitch[imageFormat.GetType()]();
+                }
+                else
+                {
+                    rawRgbData = null;
+                    rawAlpha = null;
+
+                    return false;
+                }
+
+                Configuration configuration = new Configuration();
+                ImageSharp decodedImage = decoder.Decode(configuration, new MemoryStream(imageData));
+
+                ImageFrame frame = decodedImage.Frames[0];
+
+                ImageFrame<Rgb24> frameRgb24 = frame as ImageFrame<Rgb24>;
+                if (frameRgb24 != null)
+                {
+                    GetRawDataFromRgbSource(frameRgb24, out rawRgbData);
+                    rawAlpha = null;
+
+                    return true;
+                }
+
+                ImageFrame<Rgba32> frameRgba32 = frame as ImageFrame<Rgba32>;
+                if (frameRgba32 != null)
+                {
+                    GetRawDataFromRgbaSource(frameRgba32, out rawRgbData, out rawAlpha);
+
+                    return true;
+                }
+
+                throw new NotSupportedException("Not supported image pixel format.");
+            }
+        }
+        catch (Exception ex)
+        {
+            if (ex is UnknownImageFormatException || ex is ImageProcessingException)
+            {
+                rawRgbData = null;
+                rawAlpha = null;
+                size = new TelerikSize();
```

Comment:
You can change it to the whole namespace because when you don't provide usings the clients could have difficulties finding out what TelerikSize is.